### ISSUE TITLE
Bug: [no-extraneous-class] false positive with accessor

### ISSUE THREAD

**Issue by yeonjuan** [typescript-eslint/typescript-eslint#10676]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAEDKAuZ4EtgGED2A7CivxgG8AoaaMYYAUygwCdoswBbKgLmlzuSwHMBuUp0Qpg0AGYYMACgCU0EmTJ0q8AK50sEsCAhUhAX2JGgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA99oBDJog6xktMvH7JUGSImjQO0SODABfEGqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class StaticConstants {
  accessor name: string;
  static foo() {
    return false
  }
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-extraneous-class": ["error"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

no errors

### Actual Result

```ts
class StaticConstants { // Unexpected class with only static properties. 1:7 - 1:22
  accessor name: string;
  static foo() {
    return false
  }
}
```
